tests/foreman/maintain/test_service.py

```diff
@@ -0,0 +1,267 @@
+    assert result.status == 0
+    result = sat_maintain.cli.Service.status(options={'brief': True})
+    assert 'FAIL' in result.stdout
+    assert result.status == 1
+    result = sat_maintain.cli.Service.restart()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+    result = sat_maintain.cli.Service.status(options={'brief': True})
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+
+@pytest.mark.include_capsule
+def test_positive_service_enable_disable(sat_maintain):
+    """Enable/Disable services using satellite-maintain service subcommand
+
+    :id: a0e0a052-0e21-465c-bb28-2e7613dbece6
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain service disable
+        2. Run satellite-maintain service enable
+
+    :expectedresults: Services should enable/disable
+    """
+    if not is_open('BZ:1995783'):
+        result = sat_maintain.cli.Service.disable()
+        assert 'FAIL' not in result.stdout
+        assert result.status == 0
+    result = sat_maintain.cli.Service.enable()
+    assert 'FAIL' not in result.stdout
+    assert result.status == 0
+
+
+def test_positive_foreman_service(request, sat_maintain):
+    """Validate httpd service should work as expected even stopping of the foreman service
+
+    :id: 08a29ea2-2e49-11eb-a22b-d46d6dd3b5b2
+
+    :steps:
+        1. Stop foreman service
+        2. Check the status for httpd should not have affect
+        3. Run satellite-maintain health check
+
+    :expectedresults: foreman service should restart correctly
+    """
+    result = sat_maintain.cli.Service.stop(options={'only': 'foreman'})
+    assert result.status == 0
+    assert 'FAIL' not in result.stdout
+    assert 'foreman' in result.stdout
+    result = sat_maintain.cli.Service.status(options={'only': 'httpd'})
+    assert result.status == 0
+    result = sat_maintain.cli.Health.check(options={'assumeyes': True})
+    assert result.status == 0
+    assert 'foreman' in result.stdout
+
+    @request.addfinalizer
+    def _finalize():
+        assert sat_maintain.cli.Service.start(options={'only': 'foreman'}).status == 0
+
+
+@pytest.mark.include_capsule
+def test_positive_status_clocale(sat_maintain):
+    """Test service status with C locale
+
+    :id: 143dda54-5ab5-478a-b33e-af805bace2d7
+
+    :parametrized: yes
+
+    :steps:
+        1. Run LC_ALL=C satellite-maintain service stop
+
+    :expectedresults: service status works with C locale
+    """
+    assert sat_maintain.cli.Service.status(env_var='LC_ALL=C').status == 0
+
+
+def test_positive_service_restart_without_hammer_config(missing_hammer_config, sat_maintain):
+    """Restart services using service restart when hammer config/credentials are missing
+
+    :id: c7518650-d72a-47b1-8d38-42b862f474fc
+
+    :steps:
+        1. Run satellite-maintain service restart
+
+    :expectedresults: Services restart should work even if hammer config/credentials are missing
```

Comment:
Thats weird behavior of satellite or FM, an attacker (Security) can simply run this command to shutdown the services without satellite credentials.

We should file a bug !

Reply:
f-m services will be independent to hammer credentials in `.hammer/cli.modules.d/foreman.yml`, its not related to satellite credentials :)

Reply:
Yeah! But ultimately the impact remains same. I mean if the credentials are not being entertained by F-M then it should be now.

Not blocking your Pr for this though!